packages/intl-displaynames/tests/index.test.ts

```diff
@@ -8,45 +8,142 @@ import * as zh from './locale-data/zh.json'
 DisplayNames.__addLocaleData(en, zh)
 
 describe('.of()', () => {
-  it('accepts case-insensitive language code with region subtag', () => {
-    expect(new DisplayNames('zh', {type: 'language'}).of('zh-hANs-sG')).toBe(
-      '简体中文（新加坡）'
-    )
+  describe('with type set to "language"', () => {
+    it('accepts case-insensitive language code with region subtag', () => {
+      expect(new DisplayNames('zh', {type: 'language'}).of('zh-hANs-sG')).toBe(
+        '简体中文（新加坡）'
+      )
+    })
+
+    it('preserves unrecognized region subtag in language code when fallback option is code', () => {
+      expect(
+        new DisplayNames('zh', {type: 'language', fallback: 'code'}).of(
+          'zh-Hans-Xy'
+        )
+      ).toBe('简体中文（XY）')
+    })
+
+    it('finds the long style language correctly', () => {
+      expect(
+        new DisplayNames('en', {type: 'language', style: 'long'}).of('en-GB')
+      ).toBe('British English')
+    })
+
+    it('finds the short style language correctly', () => {
+      expect(
+        new DisplayNames('en', {type: 'language', style: 'short'}).of('en-GB')
+      ).toBe('UK English')
+    })
+
+    describe('with languageDisplay set to "dialect"', () => {
+      it('finds the style language correctly', () => {
+        expect(
+          new Intl.DisplayNames('en', {
+            type: 'language',
+            languageDisplay: 'dialect',
+          }).of('en-GB')
+        ).toBe('British English')
+      })
+    })
+
+    describe('with languageDisplay set to "standard"', () => {
+      it('finds the long style language correctly', () => {
+        expect(
+          new DisplayNames('en', {
+            type: 'language',
+            style: 'long',
+            languageDisplay: 'standard',
+          }).of('en-GB')
+        ).toBe('English (United Kingdom)')
+      })
+
+      it('finds the short style language correctly', () => {
+        expect(
+          new DisplayNames('en', {
+            type: 'language',
+            style: 'short',
+            languageDisplay: 'standard',
+          }).of('en-GB')
+        ).toBe('English (UK)')
+      })
+    })
+
+    describe('with fallback set to "none"', () => {
+      it('returns undefined when called with language code that has unrecognized region subtag', () => {
+        expect(
+          new DisplayNames('zh', {type: 'language', fallback: 'none'}).of(
+            'zh-Hans-XY'
+          )
+        ).toBe(undefined)
+      })
+
+      it('returns undefined when called with language code that valid region subtag but invalid language subtag', () => {
+        expect(
+          new DisplayNames('zh', {type: 'language', fallback: 'none'}).of(
+            'xx-CN'
+          )
+        ).toBe(undefined)
+      })
+    })
   })
 
   it('accepts case-insensitive currency code', () => {
     expect(
-      new DisplayNames('en-US', {type: 'currency', style: 'long'}).of('cNy')
```

Comment:
why this change?

Reply:
Because the value of the `locale` field inside the `./locale-data/en.json` file is `en`, not `en-US`; so English localization data are bound to the `en` locale. For this reason I thought to correct the value in this test

Reply:
`en` & `en-US` canonicalize to the same thing. If you keep it as `en-US` does it crap out?

Reply:
It works with `en-US` too. Do you want me to revert it to `en-US` and update all the new tests to use `en-US`?

Reply:
I think just revert this one for sanity check. There's some locale matching mechanism underneath the hood and I wanna make sure it doesn't change that

Reply:
Done